[14.0][MIG] agreement_legal (Version 12.0 to 14.0)

patrickrwilson opened the pull request.

ygol committed:
[ADD] agreement_legal migration

ygol committed:
[FIX] flake8

ygol committed:
[FIX] flake8(2)

ygol committed:
[FIX] agreement_legal: remove account dependency

ygol committed:
[FIX] ooops

[UPD] README.rst

ygol committed:
[FIX] handling code field + partner_id

[UPD] Update agreement_legal.pot

ygol committed:
[UPD] remove field reference and use field code

Translated using Weblate (French)

Currently translated at 100.0% (301 of 301 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/fr/

[UPD] Update agreement_legal.pot

ygol committed:
[ADD] custom dynamic parties

ygol committed:
[FIX] flake8

ygol committed:
[FIX] pylint

ygol committed:
[IMP] add default html value for parties

ygol committed:
[UPD] flake8

murtuzasaleh committed:
[FIX][12.0] Menu Icon.

[UPD] Update agreement_legal.pot

max3903 committed:
[FIX] Icon in settings

max3903 committed:
[FIX] agreement_legal: module name

[UPD] Update agreement_legal.pot

Added translation using Weblate (Portuguese (Portugal))

max3903 committed:
[FIX] Issue #323 + Cleanup

max3903 committed:
Update agreement_section.py

max3903 committed:
[FIX] Duplicate label

max3903 committed:
[FIX] Clauses are copied from the sections

max3903 committed:
[FIX] agreement_legal: Copy

max3903 committed:
[FIX] View priority

Translated using Weblate (Portuguese (Portugal))

Currently translated at 2.7% (7 of 260 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt_PT/

Translated using Weblate (Portuguese (Portugal))

Currently translated at 21.9% (57 of 260 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt_PT/

max3903 committed:
[FIX] Issue #331

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/

max3903 committed:
[IMP] agreement_legal: add readonly group

[UPD] Update agreement_legal.pot

[UPD] Update agreement_legal.pot

ygol committed:
[UPD] update agreement demo data with compulsory fields

ygol committed:
[UPD] update agreement demo data with compulsory fields

ygol committed:
[UPD] move data to demo keyword

ygol committed:
[UPD] PR review's remarks

ygol committed:
Update agreement_legal/__manifest__.py

Co-Authored-By: Maxime Chambreuil <mchambreuil@opensourceintegrators.com>

[UPD] README.rst

[UPD] Update agreement_legal.pot

ygol committed:
[FIX] Issue #342

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/

Translated using Weblate (French)

Currently translated at 100.0% (260 of 260 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/fr/

[UPD] README.rst

brian10048 committed:
[IMP] agreement_serviceprofile: add stage

- Add stage_type to agreement.stage
- Track stage of service profile
- Add Kanban view for service profiles to the dashboard
- Add product_id field to service profile

[UPD] Update agreement_legal.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/

Added translation using Weblate (Chinese (Simplified))

Translated using Weblate (Chinese (Simplified))

Currently translated at 4.5% (13 of 287 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/zh_CN/

Translated using Weblate (Chinese (Simplified))

Currently translated at 39.4% (113 of 287 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/zh_CN/

Translated using Weblate (Chinese (Simplified))

Currently translated at 53.0% (152 of 287 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/zh_CN/

Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (287 of 287 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/zh_CN/

ygol committed:
Update agreement_legal/models/agreement.py

Co-Authored-By: kongrattapong <52144935+kongrattapong@users.noreply.github.com>

Translated using Weblate (French)

Currently translated at 100.0% (287 of 287 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/fr/

gurneyalex committed:
[IMP] move some notions from agreement_legal to agreement

we move the is_template field definition and the agreement.type model from
the agreement_legal module to the agreement module.

The fields are not displayed by default, unless the feature is enabled through a
technical feature group, this is configurable in the agreement_sale module (because agreement
in itself has no UI, and agreement_legal enables the feature by default)

[UPD] Update agreement_legal.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/

Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (284 of 284 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/zh_CN/

OCA-git-bot committed:
agreement_legal 12.0.1.1.1

Added translation using Weblate (Portuguese (Brazil))

Translated using Weblate (Portuguese (Brazil))

Currently translated at 54.9% (156 of 284 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt_BR/

Translated using Weblate (Portuguese (Brazil))

Currently translated at 97.5% (277 of 284 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt_BR/

Translated using Weblate (Portuguese (Portugal))

Currently translated at 33.1% (94 of 284 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt_PT/

Translated using Weblate (Portuguese (Portugal))

Currently translated at 40.8% (116 of 284 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt_PT/

marcelsavegnago committed:
[IMP] New icon for agreement modules (svg file in agreement_legal)

Translated using Weblate (Portuguese (Portugal))

Currently translated at 73.9% (210 of 284 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt_PT/

Added translation using Weblate (Portuguese)

Translated using Weblate (Portuguese)

Currently translated at 75.4% (214 of 284 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt/

Translated using Weblate (Portuguese)

Currently translated at 75.7% (215 of 284 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt/

Translated using Weblate (Portuguese)

Currently translated at 75.7% (215 of 284 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt/

Tardo committed:
[FIX] agreement_legal: Change company_contact_id domain

Before this commit, the view of agreement uses a dotted domain
on the 'right leaf' in the company_conact_id field causing
an javascript error.

Translated using Weblate (Portuguese (Portugal))

Currently translated at 73.9% (210 of 284 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt_PT/

Translated using Weblate (Portuguese)

Currently translated at 77.5% (220 of 284 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt/

[UPD] Update agreement_legal.pot

OCA-git-bot committed:
agreement_legal 12.0.1.2.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/

MiquelRForgeFlow committed:
[FIX] company_partner_id and partner_id have the same label: Partner.

[UPD] Update agreement_legal.pot

Tardo committed:
[FIX] agreement_legal: Change partner_id domain in agreement view

Previous this commit users can't select a vendor in the agreement
form, this is not the behaviour described in the help. This
commit changes the domain to allow select a vendor.

OCA-git-bot committed:
agreement_legal 12.0.1.2.1

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/

OCA-git-bot committed:
agreement_legal 12.0.1.3.0

Translated using Weblate (Portuguese)

Currently translated at 99.6% (284 of 285 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt/

OCA-git-bot committed:
agreement_legal 12.0.1.3.1

[UPD] Update agreement_legal.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/

Commit:
[IMP] agreement,agreement_legal: support activities

- The `agreement` model now features mail and activities from its base module.
- The `agreement_legal` module doesn't need to add mail support now.
- Add activities to form views.

@Tecnativa TT22477

[UPD] Update agreement_legal.pot

OCA-git-bot committed:
agreement_legal 12.0.1.4.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/

[UPD] Update agreement_legal.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/

Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (285 of 285 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt_BR/

Added translation using Weblate (Spanish)

[UPD] Update agreement_legal.pot

Translated using Weblate (Spanish)

Currently translated at 14.0% (40 of 285 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/es/

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/

Translated using Weblate (Spanish)

Currently translated at 99.3% (284 of 286 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/es/

[UPD] README.rst

[UPD] README.rst

OCA-git-bot committed:
agreement_legal 12.0.1.4.1

murtuzasaleh committed:
[FIX] agreement_legal

Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (286 of 286 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt_BR/

Translated using Weblate (Spanish)

Currently translated at 99.3% (284 of 286 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/es/

Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (286 of 286 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt_BR/

Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (286 of 286 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt_BR/

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/

Translated using Weblate (French)

Currently translated at 98.6% (282 of 286 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/fr/

Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (286 of 286 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/pt_BR/

Translated using Weblate (French)

Currently translated at 100.0% (286 of 286 strings)

Translation: contract-12.0/contract-12.0-agreement_legal
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_legal/fr/

newtratip committed:
[12.0][FIX] agreement_legal: Error when mass action on list view

OCA-git-bot committed:
agreement_legal 12.0.2.0.1

patrickrwilson committed:
[14.0][MIG] agreement_legal (Version 12.0 to 14.0)

[MIG] Black & Cleanup

[IMP] Reviewer Requested Changes

[IMP] Black

[IMP] Reviewer Suggestion

[FIX] Tests

[FIX] Multiple Issues

- Added 'is_old_version' field to better display old versions. Before these were marked as inactive therefore not showing in the agreement.
- 'is_template' field now copies when making a template revision or copying a template.
- New Versions now retain original create date.
- Moved version/revision and created on/by fields out of footer to top in order to have the save buttons when creating children.
- Moved demo data out of data folder into the demo.xml

[FIX] demo data

[FIX] Revert back to only migration

[FIX] Report dynamic field rendering

[FIX] Test

patrickrwilson commented:
@osi-scampbell @ReedHayashikawa can you assist with review?

kittiu commented:
@patrickrwilson this is the same as https://github.com/OCA/contract/pull/623 ?

cc @newtratip

patrickrwilson commented:
@kittiu Yes, it's the same. I didn't do a good job checking before submitting this so I commented on @newtratip's PR offering for him to use the tests in this PR to finish making theirs green but haven't received a response yet.

kittiu commented:
@newtratip since this one already green and it seem to be a complete migration. I think it is easier to go with this PR.

Also, as I checked your PR #623, it has enhancement, i..e, create agreement from template (wizard). So, lets push PR to merge first, and then you can add another to propose the enhancement on top of this one.

WDYT?

kittiu commented on agreement_legal/__manifest__.py:
```diff
@@ -0,0 +1,43 @@
+    "website": "https://github.com/OCA/contract",
+    "category": "Partner",
+    "license": "AGPL-3",
+    "version": "14.0.2.0.0",
```
```suggestion
    "version": "14.0.1.0.0",
```

kittiu commented on agreement_legal/data/demo.xml:
```diff
@@ -0,0 +1,124 @@
+        <function name="search" model="ir.model.data">
+            <value eval="[('module', '=', 'agreement'), ('name', '=', 'market1')]" />
+        </function>
+
```
```suggestion
```
Just the spacing corrections (there are more in this file, I wont' mentioned all).

kittiu commented on agreement_legal/data/demo.xml:
```diff
@@ -0,0 +1,124 @@
+        </function>
+        <value eval="{'noupdate': True}" />
+    </function>
+    <function name="write" model="ir.model.data">
```
```suggestion

    <function name="write" model="ir.model.data">
```

ygol replied:
hello, shouldn't "agreement_legal/data/demo.xml" only be loaded if demo data is set?

kittiu commented on agreement_legal/models/agreement_appendix.py:
```diff
@@ -0,0 +1,98 @@
+    title = fields.Char(
+        string="Title",
+        required=True,
+        help="The title is displayed on the PDF." "The name is not.",
```
```suggestion
        help="The title is displayed on the PDF. The name is not.",
```

kittiu commented on agreement_legal/models/agreement_clause.py:
```diff
@@ -0,0 +1,97 @@
+
+    name = fields.Char(string="Name", required=True)
+    title = fields.Char(
+        string="Title", help="The title is displayed on the PDF." "The name is not."
```
```suggestion
        string="Title", help="The title is displayed on the PDF. The name is not."
```

kittiu commented on agreement_legal/models/agreement_recital.py:
```diff
@@ -0,0 +1,94 @@
+    name = fields.Char(string="Name", required=True)
+    title = fields.Char(
+        string="Title",
+        help="The title is displayed on the PDF." "The name is not.",
```
```suggestion
        help="The title is displayed on the PDF. The name is not.",
```

kittiu commented on agreement_legal/models/agreement.py:
```diff
@@ -0,0 +1,412 @@
+    parties = fields.Html(
+        string="Parties",
+        tracking=True,
+        default=_get_default_parties,
```
```suggestion
        default=lambda self: self._get_default_parties(),
```
https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#38fields

kittiu commented on agreement_legal/models/agreement.py:
```diff
@@ -0,0 +1,412 @@
+    # Create New Version Button
+    def create_new_version(self):
+        for rec in self:
+            if not rec.state == "draft":
+                # Make sure status is draft
+                rec.state = "draft"
```
```suggestion
            if not rec.state == "draft":
                # Make sure status is draft
                rec.state = "draft"
```
I am not sure why we need to reset state of rec to "draft".
Or why not just put {"state": "draft"} in to default_vals.

kittiu commented on agreement_legal/models/agreement.py:
```diff
@@ -0,0 +1,412 @@
+
+    def create_new_agreement(self):
+        default_vals = {
+            "name": "NEW",
```
Shoulde be "New" or "NEW" (I saw "New" most of the time in odoo code :)

kittiu commented on agreement_legal/models/agreement.py:
```diff
@@ -0,0 +1,412 @@
+            "state": "draft",
+            "stage_id": self.env.ref("agreement_legal.agreement_stage_new").id,
+        }
+        res = self.copy(default=default_vals)
```
If no for loop, should we have `self.ensure_one()`

kittiu commented on agreement_legal/models/agreement.py:
```diff
@@ -0,0 +1,412 @@
+            vals["code"] = self.env["ir.sequence"].next_by_code("agreement") or _("New")
+        if not vals.get("stage_id"):
+            vals["stage_id"] = self.env.ref("agreement_legal.agreement_stage_new").id
+        return super(Agreement, self).create(vals)
```
```suggestion
        return super().create(vals)
```

kittiu commented on agreement_legal/models/agreement.py:
```diff
@@ -0,0 +1,412 @@
+    # Increments the revision on each save action
+    def write(self, vals):
+        vals["revision"] = self.revision + 1
+        return super(Agreement, self).write(vals)
```
```suggestion
        return super().write(vals)
```

kittiu commented on agreement_legal/static/src/xml/domain_widget_view.xml:
```diff
@@ -0,0 +1,307 @@
+<templates id="template" xml:space="preserve">
+    <t t-name="FieldDomain.content">
+        <t t-if="partialUse">
+            <div t-if="hasModel" class="o_field_domain_panel">
```
Should we remove this commented section?

ygol replied:
hello, shouldn't "agreement_legal/data/demo.xml" only be loaded if demo data is set?

patrickrwilson commented:
@kittiu Thank you for your review, your suggestions have been implemented.

kittiu commented:
> @kittiu Thank you for your review, your suggestions have been implemented.

Thanks. Please check travis.

patrickrwilson commented:
@ygol it's ready for your review.

ygol commented:
> @ygol it's ready for your review.

I've functionally tested. Those are my findings:

- revision doesn't seem to work. (doesn't list previous versions)

- new version of template loose the "is_template" boolean

- Problem with reference when duplicating an agreement (also when creating from a template).
cf. https://github.com/OCA/contract/issues/640

- problem with "created on" when generating a new version. the archived version get the current date and the new version keeps the origin date (should be switched).
cf. https://github.com/OCA/contract/issues/502

- when adding a child agreement from parent there is no way to save the created child AG + fields should be pre-populated with values from the parent ("parent agreement", "start date", "end date", contacts, etc.)
cf. https://github.com/OCA/contract/issues/500

- I think "data/demo.xml" should only be loaded when demo data is required  https://github.com/OCA/contract/pull/669/files#diff-289fece69d3e767eb82c1fa91fff8a6ae1b74aa76123ef49d0c1aaf49532c9a0R21

patrickrwilson commented:
@ygol  my last commit has most of your fixes, see my notes below....

* revision doesn't seem to work. (doesn't list previous versions) - FIXED: I added a new field called 'is_old_version' which it now uses to show old versions. Before it was marking them as inactive which is why they weren't visible.
* new version of template loose the "is_template" boolean - FIXED: The agreement module this inherits has copy=False for the is_template fields so in agreement_legal I set it to True
* Problem with reference when duplicating an agreement (also when creating from a template). #640 - This is part of the Agreement module this inherits so I think any changes need to be done in that module.
* problem with "created on" when generating a new version. the archived version get the current date and the new version keeps the origin date (should be switched). #502 - FIXED: I've made it to retain the original create date/time.
* when adding a child agreement from parent there is no way to save the created child AG + fields should be pre-populated with values from the parent ("parent agreement", "start date", "end date", contacts, etc.) #500 - FIXED: I've added context to the one2many field.
* I think "data/demo.xml" should only be loaded when demo data is required  https://github.com/OCA/contract/pull/669/files#diff-289fece69d3e767eb82c1fa91fff8a6ae1b74aa76123ef49d0c1aaf49532c9a0R21 - FIXED: I've removed the items in data/demo.xml since they are already in demo/demo.xml

patrickrwilson commented:
It looks like I have to fix a test that now fails after these changes. I will work on that shortly.

ygol commented:
@patrickrwilson thanks so much. here updated remarks. Are those opportune or is it better to first do the migration and leave those fixing for a later PR?

> * revision doesn't seem to work. (doesn't list previous versions) - FIXED: I added a new field called 'is_old_version' which it now uses to show old versions. Before it was marking them as inactive which is why they weren't visible.

Wouldn't be possible to use "context={'active_test': False}" to show those archived revisions? I think there is no way to list "old_version" beside in the history tab now.

> * new version of template loose the "is_template" boolean - FIXED: The agreement module this inherits has copy=False for the is_template fields so in agreement_legal I set it to True

Now when you click on "new agreement" button it creates a template. so if a tradeoff must be done; IMO better to have the original templates loosing their "is_template" when copying. 


> * problem with "created on" when generating a new version. the archived version get the current date and the new version keeps the origin date (should be switched). #502 - FIXED: I've made it to retain the original create date/time.

IMO the date of a version should be the date that version has been created. so if original version was created on 4.5.21 and a new version created on 6.8.21 then version on date would show 4.5.21 and version 2 would show 6.8.21. 
Any other opinion on this?

patrickrwilson commented:
@ygol I agree that these changes should be in a later PR. The migration should probably migrate it as it was and then list of improvements/fixes made in additional PR's, that way they can properly be vetted and not grouped into the migration task. I will revert the changes back and then we can perform the changes after the migration is merged.

patrickrwilson commented:
@ygol reverted changes and squashed, should be ready for review and merge, then we can look at improving/fixing those issues.

ygol commented:
There's a problem with the report -:(
1. go to an agreement
2. click on preview or print
3. unwanted characters appears in the report

![image](https://user-images.githubusercontent.com/2945174/123448626-c6879980-d5e3-11eb-9b99-184884cb0775.png)

ygol commented:
> There's a problem with the report -:(
> 
> 1. go to an agreement
> 2. click on preview or print
> 3. unwanted characters appears in the report
> 
I think that v14 uses now 'mail_render_mixin' and following 
https://github.com/odoo/odoo/blob/70a75fa1b41f06c10abe02efbb0b4565b4190e45/addons/mail/models/mail_render_mixin.py#L388
it seem a dict is returned. So for example; adding the following line 299 in agreement.py
```
    def _compute_dynamic_description(self):
        MailTemplates = self.env["mail.template"]
        for agreement in self:
            lang = agreement.partner_id.lang or "en_US"
            description = MailTemplates.with_context(lang=lang)._render_template(
                agreement.description, "agreement", [agreement.id]
            )
            des = ""
            for i in description:
                des += description[i]
            agreement.dynamic_description = des
```
seems to fix it.
There's an easier way maybe?

patrickrwilson commented:
@ygol I've used your suggested code which worked. I'm not sure there is an easier way but this fixed the issue which I'm good with. Hopefully it's good now to merge.

hkapatel-initos commented:
LGTM Can we please merge this PR?

newtratip commented:
@patrickrwilson, CI is red, Can you check please ?

newtratip commented:
@OCA/project-service-maintainers, I think this PR should be merge. I have a lot of enhancement according https://github.com/OCA/contract/pull/623. I will open in another PR.

Thanks!

simahawk commented:
/ocabot merge nobump

newtratip commented:
@patrickrwilson, I install this module, it's throw error. `Cannot update missing record 'agreement.market1'`
![Selection_268](https://user-images.githubusercontent.com/24691983/124267948-1c6bbc80-db63-11eb-913a-935e7f7390e3.png)

Create database, not demo data

Why do you add data file https://github.com/OCA/contract/pull/669/files#diff-289fece69d3e767eb82c1fa91fff8a6ae1b74aa76123ef49d0c1aaf49532c9a0R21 ? I saw demo data  in demo folder and data folder, it's same ?

patrickrwilson commented:
@newtratip Please see new PR #696 that removes that data/demo.xml which shouldn't be there and fixes this installation error. Your review would be appreciated.

Thanks
